File: solutions/php/binary-search/1/BinarySearch.php
<?php
declare(strict_types=1);

function find(int $needle, array $haystack): int
{
    $left = 0;
    $right = count($haystack) - 1;
    
    while ($left <= $right) {
        $mid = (int)floor(($left + $right) / 2);
        
        if ($haystack[$mid] === $needle) {
            return $mid;
        }
        
        if ($haystack[$mid] < $needle) {
            $left = $mid + 1;
        } else {
            $right = $mid - 1;
        }
    }
    
    return -1;
}
?>

File: solutions/ruby/binary-search/1/binary_search.rb
class BinarySearch
  def initialize(list)
    @list = list
  end

  # Cambiamos 'find' por 'search_for' para coincidir con los tests
  def search_for(target)
    left = 0
    right = @list.length - 1

    while left <= right
      mid = (left + right) / 2
      middle_element = @list[mid]

      if middle_element == target
        return mid
      elsif middle_element < target
        left = mid + 1
      else
        right = mid - 1
      end
    end

    nil
  end
end